### DISCUSSION TITLE
Variable interaction plot with high interactions

### DISCUSSION THREAD

**Discussion by guvensimsek** [kogalur/randomForestSRC#218] [Q&A] [answered]
![interactionplot](https://user-images.githubusercontent.com/94964130/158754774-39692932-8165-439e-bdd9-c18f9a3663e2.jpg)

Is it normal for such high interaction between the variables used in the dataset? Is there a solution to this?

**Comment by ishwaran** [comment, marked as answer]
Interactions can happen and random forests is very good at fitting interactions.  For example, here's the R code for running a partial analysis of the air quality data that shows a strong interaction between temperature and wind (from the [partial.rfsrc helpfile](https://luminwin.github.io/randomForestSRC/reference/partial.rfsrc.html))

## ------------------------------------------------------------
##
## regression: partial effects for two variables simultaneously
##
## ------------------------------------------------------------

airq.obj <- [rfsrc](https://luminwin.github.io/randomForestSRC/reference/rfsrc.html)(Ozone ~ ., data = airquality)

## specify wind and temperature values of interest
wind <- [sort](https://rdrr.io/r/base/sort.html)([unique](https://rdrr.io/r/base/unique.html)(airq.obj$xvar$Wind))
temp <- [sort](https://rdrr.io/r/base/sort.html)([unique](https://rdrr.io/r/base/unique.html)(airq.obj$xvar$Temp))

## partial effect for wind, for a given temp
pdta <- [do.call](https://rdrr.io/r/base/do.call.html)(rbind, [lapply](https://rdrr.io/r/base/lapply.html)(temp, function(x2) {
  o <- partial(airq.obj,
         partial.xvar = "Wind", partial.xvar2 = "Temp",
         partial.values = wind, partial.values2 = x2)
  [cbind](https://rdrr.io/r/base/cbind.html)(wind, x2, get.partial.plot.data(o)$yhat)
}))
pdta <- [data.frame](https://rdrr.io/r/base/data.frame.html)(pdta)
[colnames](https://rdrr.io/r/base/colnames.html)(pdta) <- [c](https://rdrr.io/r/base/c.html)("wind", "temp", "effectSize")

## [coplot](https://rdrr.io/r/graphics/coplot.html) of partial effect of wind and temp 
coplot(effectSize ~ wind|temp, pdta, pch = 16, overlap = 0)
